File: dcmcuong/VIETBAIT.DICOM.BASE/DataElement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace VIETBAIT.DICOM.BASE
{
    public class DataElement
    {
        #region Public Property
        public byte[] DataValue {get; set;}
        public UInt32 DataTag {get; set; }
        public UInt32 Length {get; set;}
        public DataElement Parent { get; set; }
        private DataElement _parent;
        #endregion

        #region Contructor

        public DataElement()
        {
           
        }
        public DataElement (UInt32 tag,UInt32 len)
        {
            DataTag = tag;
            Length = len;
        }

        public DataElement (UInt32 tag,byte [] datavalue)
        {
            DataTag = tag;
            DataValue = datavalue;
            Length = (UInt32 ) datavalue.Length;
        }

        #endregion

        #region Destructor

         ~DataElement ()
        {
            DataValue = null;
        }

        #endregion


        public  byte[] CreateByteBuff()
        {
            var byteBuff = new byte[8];
            byte[] tmp = BitConverter.GetBytes((UInt16)(DataTag >> 16));
            Array.Copy(tmp, 0, byteBuff, 0, 2);
            tmp = BitConverter.GetBytes((UInt16)(DataTag & 0xFFFF));
            Array.Copy(tmp, 0, byteBuff, 2, 2);
            tmp = BitConverter.GetBytes(Length);
            Array.Copy(tmp, 0, byteBuff, 4, tmp.Length);
            if ((Length != 0xFFFFFFFFUL) && (Length > 0))
            {                
                Array.Resize<byte>(ref byteBuff, (int) (Length + 8));
                DataValue.CopyTo(byteBuff, 8);

            }
            return byteBuff;
        }
    }
}

File: dcmcuong/VIETBAIT.DICOMPDU/DataElement.vb

Public Class DataElement
    Dim DataTagValue As UInt32
    Dim TagGroupValue As UInt16
    Dim TagElementValue As UInt16
    Dim ValueLength As UInt32
    Public DataValue() As Byte


    Public Property DataTag() As UInt32
        Get
            Return DataTagValue
        End Get
        Set(ByVal value As UInt32)
            DataTagValue = value
            TagGroupValue = CUShort(value >> 16)
            TagElementValue = CUShort(value And &HFFFFUI)
        End Set
    End Property

    Public Property TagGroup() As UInt16
        Get
            Return TagGroupValue
        End Get
        Set(ByVal value As UInt16)
            TagGroupValue = value
            DataTagValue = (DataTagValue And &HFFFFUI) + (CUInt(value) << 16)
        End Set
    End Property

    Public Property TagElement() As UInt16
        Get
            Return TagElementValue
        End Get
        Set(ByVal value As UInt16)
            TagElementValue = value
            DataTagValue = (DataTagValue And &HFFFF0000UI) + value
        End Set
    End Property

    Public Property Length() As UInt32
        Get
            Return ValueLength
        End Get
        Set(ByVal value As UInt32)
            ValueLength = value
        End Set
    End Property

    Sub SetLength()
        If DataValue Is Nothing Then
            ValueLength = 0
        Else
            ValueLength = DataValue.Length
        End If

    End Sub

    Function CreateByteBuff() As Byte()
        Dim ByteBuff(7) As Byte
        Dim tmp() As Byte
        tmp = BitConverter.GetBytes(TagGroup)
        [Array].Copy(tmp, 0, ByteBuff, 0, 2)
        tmp = BitConverter.GetBytes(TagElement)
        [Array].Copy(tmp, 0, ByteBuff, 2, 2)
        tmp = BitConverter.GetBytes(ValueLength)
        [Array].Copy(tmp, 0, ByteBuff, 4, tmp.Length)
        If (ValueLength = &HFFFFFFFFUL) Then
            ValueLength = 0
        End If
        If (ValueLength > 0) Then
            Array.Resize(ByteBuff, MyClass.Length + 8)
            [Array].Copy(DataValue, 0, ByteBuff, 8, ValueLength)
        End If
        Return ByteBuff
    End Function
End Class